File: backend/src/models/Customer.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface ICustomer extends Document {
  name: string;
  email: string;
  phone: string;
  status: 'Active' | 'Settled' | 'Disputed' | 'Escalated' | 'Inactive';
  assignedOfficer?: mongoose.Types.ObjectId;
  googleSheetRowId?: number;
  createdAt: Date;
  updatedAt: Date;
}

const CustomerSchema: Schema = new Schema(
  {
    name: { type: String, required: true },
    email: { type: String, required: true, unique: true, index: true },
    phone: { type: String, required: true },
    status: {
      type: String,
      enum: ['Active', 'Settled', 'Disputed', 'Escalated', 'Inactive'],
      default: 'Active',
      index: true,
    },
    assignedOfficer: { type: Schema.Types.ObjectId, ref: 'User' },
    googleSheetRowId: { type: Number },
  },
  { timestamps: true }
);

export default mongoose.model<ICustomer>('Customer', CustomerSchema);

File: backend/models/customer.py
from pydantic import BaseModel, EmailStr
from typing import Optional
from datetime import datetime


class CustomerCreate(BaseModel):
    name: str
    email: EmailStr
    phone: str
    outstanding_amount: float = 0
    days_overdue: int = 0
    assigned_officer: Optional[str] = None


class CustomerUpdate(BaseModel):
    status: Optional[str] = None  # Active | Settled | Disputed | Escalated | Inactive
    assigned_officer: Optional[str] = None


class CustomerResponse(BaseModel):
    id: str
    name: str
    email: str
    phone: str
    status: str
    assigned_officer: Optional[dict] = None
    google_sheet_row_id: Optional[int] = None
    outstanding_amount: float = 0
    days_overdue: int = 0
    risk_score: int = 0
    has_active_plan: bool = False
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
